src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3645,74 +3693,114 @@ protected override void VisitBinaryOperatorChildren(ArrayBuilder<BoundBinaryOper
                 binary = stack.Pop();
             }
 
-            // learn from true/false constant
-            bool learnFromBooleanConstantTest()
+            static bool isEqualsOrNotEquals(BoundBinaryOperator binary)
+                => binary.OperatorKind.Operator() is BinaryOperatorKind.Equal or BinaryOperatorKind.NotEqual;
```

Comment:
Honestly feels like these should be inlined. I was expecting the logic to be more complicated because you broke it out into a separate function, but it's actually a pretty simple check.

Reply:
I like having `isEquals` as its own method because several calls to it are operands of `==` and it feels a little cleaner than parenthesizing a nested `==`, for example. But I will go ahead and eliminate `isEqualsOrNotEquals`.